scripts/provision_credentials.py

```diff
@@ -0,0 +1,141 @@
+import base64
+import json
+import logging
+import os
+import sys
+import uuid
+from functools import partial
+
+import boto3
+import googleapiclient.discovery
+from botocore.exceptions import ClientError
+from google.oauth2 import service_account
+from googleapiclient.errors import HttpError
+
+from azul import config
+
+logger = logging.getLogger(__name__)
+logging.basicConfig(level=logging.INFO)
+
+
+def parse_google_key(response):
+    return base64.decodebytes(bytes(response['privateKeyData'], 'ascii')).decode()
+
+
+def get_google_service():
+    credentials = service_account.Credentials.from_service_account_file(
+        filename=os.environ['GOOGLE_APPLICATION_CREDENTIALS'],
+        scopes=['https://www.googleapis.com/auth/cloud-platform'])
+    return googleapiclient.discovery.build('iam', 'v1', credentials=credentials)
+
+
+def create_service_account_creds(service_account_email):
+    service = get_google_service()
+    key = service.projects().serviceAccounts().keys().create(
+        name='projects/-/serviceAccounts/' + service_account_email, body={}
+        ).execute()
+    logger.info("Successfully created service account key for user '%s'", service_account_email)
+    return parse_google_key(key)
+
+
+def destroy_service_account_creds(client, service_account_email):
+    try:
+        creds = client.get_secret_value(SecretId=config.secrets_manager_secret_name('google_service_account'))
+    except ClientError as e:
+        if e.response['Error']['Code'] == 'ResourceNotFoundException':
+            logger.info('Secret already deleted, cannot get key_id for %s', service_account_email)
+            return
+        else:
+            raise
+    else:
+        key_id = json.loads(creds['SecretString'])['private_key_id']
+        service = get_google_service()
+        try:
+            service.projects().serviceAccounts().keys().delete(
+                name='projects/-/serviceAccounts/' + service_account_email + '/keys/' + key_id).execute()
+        except HttpError as e:
+            if e.resp.reason != 'Not Found':
+                raise
+        logger.info("Successfully deleted service account key with id '%s' for user '%s'",
+                    key_id, service_account_email)
+
+
+def random_hmac_key():
+    # Even though an hmac key can be any sequence of bytes, we restrict to base64 in order to encode as string
+    # 48 because this works out a string of length 64
+    key = base64.encodebytes(os.urandom(48)).replace(b'=', b'').replace(b'\n', b'').decode()
```

Comment:
```suggestion
    key = base64.encodebytes(os.urandom(48)).decode().replace('=', '').replace('\n', '')
```

Assert the expected length afterwards. That would also eliminate/reduce the need for the comment above.

Reply:
Good idea.